File: Prime-Number/is_prime.erl
-module(isPrime).
-export([isPrime/1]).

isPrime(0)-> false;
isPrime(1)-> false;
isPrime(2)-> true;

isPrime(N) ->
    case (N rem 2) =:= 0 of
        true -> true;
        _ -> false
    end.

File: Prime-Number/isPrime.hs
isPrime :: Int -> Bool
isPrime n = isDividable n (n-1)
    where isDividable 1 _ = False
          isDividable 2 _ = True
          isDividable x 2 = (x `mod` 2) > 0
          isDividable x y = (x `mod` y) > 0 && isDividable x (y - 1)

